CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+
+option(BUILD_TESTS "Build tests" OFF)
+option(BUILD_BENCHMARKS "Build benchmarks" OFF)
+option(USE_VCPKG "Use vcpkg for dependencies" OFF)
+
+if(COMMAND cmake_policy)
+    cmake_policy(SET CMP0003 NEW)
+endif(COMMAND cmake_policy)
+
+if (USE_VCPKG)
+    set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake
+    CACHE STRING "Vcpkg toolchain file")
+
+    if (BUILD_TESTS)
+        list(APPEND VCPKG_MANIFEST_FEATURES "tests")
+    endif()
+
+    if (BUILD_BENCHMARKS)
+        list(APPEND VCPKG_MANIFEST_FEATURES "benchmarks")
+    endif()
+endif()
+
+
+project(cachemere)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+find_package(Boost REQUIRED)
+
+add_library(cachemere INTERFACE)
+
+if ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
+  # using Clang or GCC
+  add_compile_options("-Wall" "-Wextra")
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  # using MSVC
+  add_compile_options("/W4")
+endif()
+
+target_link_libraries(cachemere
+    INTERFACE
+        Boost::boost
+)
+
+target_include_directories(cachemere
+    INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+        $<INSTALL_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+)
+
+target_sources(cachemere
```

Comment:
[detail] Since the arguments of this comment are verbose, I'd put this command last, that is after

```cmake
if (BUILD_BENCHMARKS)
    add_subdirectory(benchmarks)
endif()

if (BUILD_TESTS)
    add_subdirectory(tests)
endif()
```